Update README with new messaging (#17200)
ballerina-platform/ballerina-lang: README.md

    [![codecov](https://codecov.io/gh/ballerina-platform/ballerina-lang/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/ballerina-lang)
  [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)

> In the cloud, applications interact with their environment over the network. Ballerina is a general-purpose programming language with features optimized for these network distributed applications. It is easy to write and modify and is suitable for application programmers.

Reviewer: ```suggestion
In the cloud, applications interact with their environment over the network. Ballerina is a general-purpose programming language with features optimized for these network-distributed applications. It is easy to write and modify and is suitable for application programmers.
```

Author: I consulted Rebecca and she said it can be written both ways. Since our tagline and most materials say network distributed (without the hyphen), shall we keep as is?

Reviewer: You can write it in both ways when the term "network-distributed" is used on its own. Here, since it is used as an adjective to describe "applications", we need to say "network-distributed applications."

Anyway, I am -1 to use it as our tagline as it linguistically doesn't sound right to say "network distributed". We do get such a word even when we do a Google search. The correct term is "distributed networks".  I doubt whether it should be "distributed-network applications". To be on the safe side, we can say "applications distributed over a network" to avoid distorting the actual meaning.

#### Services

> It has first-class language constructs for providing and consuming services,

Reviewer: ```suggestion
It has first-class language constructs for providing and consuming services.
```

#### Sequence diagrams

> The structure of functions and programs allows them to be viewed as sequence diagrams showing the interactions between concurrently executing parts.

Reviewer: ```suggestion
The structure of the functions and programs allows them to be viewed as sequence diagrams showing the interactions between the components that execute concurrently.
```

Author: - First edit: This isn't referring to any functions or programs, it's referring to all. Suggest not to add 'the'.
- Second edit: Suggest to keep as is, because this is the messaging reviewed by the Ballerina team.

Reviewer: - First edit: You need to use "the" before a noun.
- It shouldn't be reviewed linguistically by the Ballerina team as if done it should be done by myself. Saying "concurrently-executing parts" is just shorthand (we need the hyphen there). It is better to say it in the proper way as "the components that execute concurrently."

#### Structural typing

> It allows for looser coupling between distributed components and eliminates the friction of data binding.

Reviewer: ```suggestion
It allows looser coupling between distributed components and eliminates the friction of data binding.
```

Author: "allows" and "allows for" hold two different meanings.  Since this messaging has been approved already shall we keep it as it is?

Reviewer: I am not sure who approved it. For me, it is right to say, "It allows looser coupling " as the meaning becomes straightforward that Ballerina allows it.
